engine/flink/minicluster/src/main/scala/pl/touk/nussknacker/engine/flink/minicluster/FlinkMiniClusterFactory.scala

```diff
@@ -102,8 +105,19 @@ class FlinkMiniClusterWithServices(
     streamExecutionEnvironmentFactory: Boolean => StreamExecutionEnvironment
 ) extends AutoCloseable {
 
-  def createStreamExecutionEnvironment(attached: Boolean): StreamExecutionEnvironment =
-    streamExecutionEnvironmentFactory(attached)
+  def withDetachedStreamExecutionEnvironment[T](action: StreamExecutionEnvironment => T): T = {
+    createStreamExecutionEnvironment(attached = false).use(env => IO(action(env))).unsafeRunSync()
+  }
+
+  // TODO: attached variant shouldn't be used. It blocks current thread infinitely during execution.
```

Comment:
The attached variant is currently used only in scenario testing and state verification code. I'm planning to change it in next PR. Thanks to that we'll avoid resource leaking which is now theoretically possible.